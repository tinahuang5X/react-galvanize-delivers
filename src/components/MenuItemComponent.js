import React from 'react';

/*
item {
  imagePath,
  name,
  price
}
*/

export default function MenuItemComponent({ item, onAddItem }) {
  if (!item) return <h5>"No Matches"</h5>;

  function handleClick(event) {
    event.preventDefault();
    onAddItem(item.id);
  }

  return (
    <div className="card MenuItemComponent">
      <div className="card-image">
        <img src={item.imagePath} alt={item.name} />
      </div>
      <div className="card-content">
        <h5 className="card-title">
          {item.name || 'N/A'}
        </h5>

        <p className="card-price">
          {item.price && typeof item.price === 'number'
            ? `$${item.price.toFixed(2)}`
            : 'N/A'}
        </p>
      </div>
      <div className="card-action">
        <a href="." className="testClick" onClick={handleClick}>
          ADD TO ORDER
        </a>
      </div>
    </div>
  );
}
