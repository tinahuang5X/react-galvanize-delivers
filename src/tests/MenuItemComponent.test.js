// Write a unit test for MenuItemComponent using Jest/Enzyme which validates the following (at minimum):
//
//     The item's name gets rendered
//     The item's price gets rendered
//     The onAddItem callback gets triggered when you click on the "Add to Order" button/link
//     Note: Shallow rendering should be sufficient
//
// Also, write a snapshot test for MenuItemComponent using Jest.

import React from 'react';
import { mount, shallow } from 'enzyme';
import MenuItemComponent from '../components/MenuItemComponent';

describe('a unit test for MenuItemComponent', () => {
  const item = {
    name: 'Some Item A',
    price: 9.99,
    imagePath: '//via.placeholder.com/300x200'
  };
  const onAddItem = jest.fn();
  const shallowWrapper = shallow(
    <MenuItemComponent item={item} onAddItem={onAddItem} />
  );

  it('item name gets rendered', () => {
    expect(shallowWrapper.find('.card-title').text()).toEqual(`Some Item A`);
  });

  it('item price gets rendered', () => {
    expect(shallowWrapper.find('.card-price').text()).toEqual('$9.99');
  });

  it('onAddItem function was triggered', () => {
    mount(<MenuItemComponent onAddItem={onAddItem} item={item} />)
      .find('.testClick')
      .simulate('click');
    expect(onAddItem).toHaveBeenCalled();
  });

  it('should pass snapshot test', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
