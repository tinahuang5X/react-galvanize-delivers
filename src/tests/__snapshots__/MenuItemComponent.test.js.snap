// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a unit test for MenuItemComponent should pass snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="card MenuItemComponent"
>
    <div
        className="card-image"
    >
        <img
            alt="Some Item A"
            src="//via.placeholder.com/300x200"
        />
    </div>
    <div
        className="card-content"
    >
        <h5
            className="card-title"
        >
            Some Item A
        </h5>
        <p
            className="card-price"
        >
            $9.99
        </p>
    </div>
    <div
        className="card-action"
    >
        <a
            className="testClick"
            href="."
            onClick={[Function]}
        >
            ADD TO ORDER
        </a>
    </div>
</div>,
  "nodes": Array [
    <div
      className="card MenuItemComponent"
>
      <div
            className="card-image"
      >
            <img
                  alt="Some Item A"
                  src="//via.placeholder.com/300x200"
            />
      </div>
      <div
            className="card-content"
      >
            <h5
                  className="card-title"
            >
                  Some Item A
            </h5>
            <p
                  className="card-price"
            >
                  $9.99
            </p>
      </div>
      <div
            className="card-action"
      >
            <a
                  className="testClick"
                  href="."
                  onClick={[Function]}
            >
                  ADD TO ORDER
            </a>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MenuItemComponent
        item={
                Object {
                        "imagePath": "//via.placeholder.com/300x200",
                        "name": "Some Item A",
                        "price": 9.99,
                      }
        }
        onAddItem={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "imagePath": "//via.placeholder.com/300x200",
            "name": "Some Item A",
            "price": 9.99,
          },
          "onAddItem": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="card MenuItemComponent"
>
          <div
                    className="card-image"
          >
                    <img
                              alt="Some Item A"
                              src="//via.placeholder.com/300x200"
                    />
          </div>
          <div
                    className="card-content"
          >
                    <h5
                              className="card-title"
                    >
                              Some Item A
                    </h5>
                    <p
                              className="card-price"
                    >
                              $9.99
                    </p>
          </div>
          <div
                    className="card-action"
          >
                    <a
                              className="testClick"
                              href="."
                              onClick={[Function]}
                    >
                              ADD TO ORDER
                    </a>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="card MenuItemComponent"
>
          <div
                    className="card-image"
          >
                    <img
                              alt="Some Item A"
                              src="//via.placeholder.com/300x200"
                    />
          </div>
          <div
                    className="card-content"
          >
                    <h5
                              className="card-title"
                    >
                              Some Item A
                    </h5>
                    <p
                              className="card-price"
                    >
                              $9.99
                    </p>
          </div>
          <div
                    className="card-action"
          >
                    <a
                              className="testClick"
                              href="."
                              onClick={[Function]}
                    >
                              ADD TO ORDER
                    </a>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MenuItemComponent
    item={
        Object {
            "imagePath": "//via.placeholder.com/300x200",
            "name": "Some Item A",
            "price": 9.99,
          }
    }
    onAddItem={[Function]}
/>,
}
`;
